milestone2.ts

```diff
@@ -83,34 +83,43 @@ const userData: Volunteer[] = [
   Hint: user data is stored in the userData object above. 
 */
 type GetNumber = (data: Volunteer[]) => number;
-let findAverage: GetNumber; // Define lambda function here
+let findAverage: GetNumber = (data) => {
+  const sum = data.reduce((previousValue, currentValue) => previousValue + currentValue.age,0);
+  return sum / (data.length);
+}; // Define lambda function here
 
-//console.log(findAverage(userData));
+console.log(findAverage(userData));
 
 // Question 2: Data Handling
 /* Use data handling function(s) to find the first index of someone from San Francisco (SF).
   Return -1 if no one is from San Francisco.
   Hint: Use a lambda function as a value
 */
-let findIndexAns: GetNumber; // Code here
+let findIndexAns: GetNumber = (data) => {
+  return data.findIndex((volun) => volun.city === "SF");
+}; // Code here
 
-//console.log(findIndexAns(userData));
+console.log(findIndexAns(userData));
 
 // Question 3: Filtering data
 /* Use data handling function(s) to find all of the people from California (CA) over an age threshold n
  */
 type GetVolunteers = (data: Volunteer[], minAge: number) => Volunteer[];
-let findCAOverN: GetVolunteers; // Code here
+let findCAOverN: GetVolunteers = (data, minAge) => {
+  return data.filter((volun) => volun.age > minAge).filter((volun) => volun.state === "CA");
```

Comment:
You can shorten this a bit with a logical and (&&)
```suggestion
  return data.filter((volun) => volun.age > minAge &&(volun) => volun.state === "CA" )
```